src/coreclr/interpreter/ee_il_dll.cpp

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#include <stddef.h>
+#include "corjit.h"
+
+#include "interpreter.h"
+#include "ee_il_dll.hpp"
+
+#include <string.h>
+#include <stdio.h>
+
+#ifdef _MSC_VER
+#define DLLEXPORT
+#else
+#define DLLEXPORT __attribute__ ((visibility ("default")))
+#endif // _MSC_VER
+
+#include <vector>
+
+// FIXME We will probably end up not needing this table.
+// If deemed useful, use some hashtable implementation instead.
+std::vector<std::pair<CORINFO_METHOD_HANDLE,InterpMethod*>> g_interpCodeHash;
+
+static InterpMethod* InterpGetInterpMethod(CORINFO_METHOD_HANDLE methodHnd)
+{
+    // FIXME lock for multiple thread access
+    for (int i = 0; i < g_interpCodeHash.size(); i++)
+    {
+        if (g_interpCodeHash[i].first == methodHnd)
+        {
+            return g_interpCodeHash[i].second;
+        }
+    }
+
+    InterpMethod* pMethod = new InterpMethod();
+    pMethod->methodHnd = methodHnd;
+
+    g_interpCodeHash.push_back({methodHnd, pMethod});
+    return pMethod;
+}
+
+/*****************************************************************************/
+ICorJitHost* g_interpHost        = nullptr;
+bool         g_interpInitialized = false;
+/*****************************************************************************/
+extern "C" DLLEXPORT void jitStartup(ICorJitHost* jitHost)
+{
+    if (g_interpInitialized)
```

Comment:
should we check and throw if someone calls startup again with a different jit host?